age_checker.module

```diff
@@ -1,394 +1,196 @@
-//               'blank_err_message' => variable_get('age_checker_' . $language_code . '_blank_error_msg', ''),
-//               'dateformat_error'  => variable_get('age_checker_' . $language_code . '_dateformat_error_msg', ''),
-//               'under_age_err_msg' => variable_get('age_checker_' . $language_code . '_underage_error_msg', ''),
-//               'date_range_err_msg' => variable_get('age_checker_' . $language_code . '_daterange_error_msg', ''),
-//               'destination' => $destination,
-//               'verify' => $verify,
-//               'day_placeholder' => variable_get('age_checker_' . $selected_country . '_day_placeholder'),
-//               'month_placeholder' => variable_get('age_checker_' . $selected_country . '_month_placeholder'),
-//               'year_placeholder' => variable_get('age_checker_' . $selected_country . '_year_placeholder'),
-//               'redirecturl' => variable_get('age_checker_under_age_url'),
-//               'id_1' => $id[0],
-//               'id_2' => $id[1],
-//               'id_3' => $id[2],
-//             ),
-//           ), 'setting');
-//         }
-//       }
-//     }
-//   }
-// }
-
-// /**
-//  * Get the ID for date element.
-//  */
-// function age_checker_get_id($key) {
-
-//   if ($key == 0) {
-//     return "#age_checker_day";
-//   }
-//   elseif ($key == 1) {
-//     return "#age_checker_month";
-//   }
-//   else {
-//     return "#age_checker_year";
-//   }
-// }
-
-// /**
-//  * Calculate visibility of age checker if set.
-//  *
-//  * Function copy from block.module, thanks for the original code.
-//  *
-//  * return boolean
-//  */
-// function age_checker_visibility() {
-
-//   $visibility = variable_get('age_checker_visibility', AGE_CHECKER_VISIBILITY_NOTLISTED);
-//   $pages = variable_get('age_checker_pages');
-
-//   // Convert path to lowercase.
-//   $pages = drupal_strtolower($pages);
-//   if ($visibility < 2) {
-//     // Convert the Drupal path to lowercase.
-//     $path = drupal_strtolower(drupal_get_path_alias(current_path()));
-//     // Compare the lowercase internal and lowercase path alias (if any).
-//     $page_match = drupal_match_path($path, $pages);
-
-//     if ($path != current_path()) {
-//       $page_match = $page_match || drupal_match_path(current_path(), $pages);
-//     }
-
-//     $page_match = !($visibility xor $page_match);
-//   }
-//   elseif (module_exists('php')) {
-//     $page_match = php_eval($pages);
-//   }
-//   else {
-//     $page_match = FALSE;
-//   }
-//   return $page_match;
-// }
+function age_checker_theme($existing, $type, $theme, $path) {
+  $age_checker_footer_links = array();
+  $age_checker_copyright = array();
+  $default_settings = \Drupal::config('age_checker.settings');
+
+  $languages = \Drupal::state()->get('age_checker_language') ? \Drupal::state()->get('age_checker_language') : $default_settings->get('age_checker_language');
```

Comment:
$language = \Drupal::state()->get('age_checker_language');
$language = $language ? $language : $default_settings->get('age_checker_language');

This would save calling \Drupal::state() twice.

or

$language = NULL;
if (empty($language = \Drupal::state()->get('age_checker_language'))) {
  $language = $default_settings->get('age_checker_language');
}